router/delivery/http/router_handler.go

```diff
@@ -109,23 +113,23 @@ func (a *RouterHandler) GetOptimalQuote(c echo.Context) (err error) {
 	tokenIn.Denom = chainDenoms[0]
 	tokenOutDenom = chainDenoms[1]
 
-	routerOpts := []domain.RouterOption{}
-
-	// Disable split routes if singleRoute is true
-	if isSingleRoute {
+	var routerOpts []domain.RouterOption
+	if req.SingleRoute {
 		routerOpts = append(routerOpts, domain.WithMaxSplitRoutes(domain.DisableSplitRoutes))
 	}
 
-	quote, err := a.RUsecase.GetOptimalQuote(ctx, tokenIn, tokenOutDenom, routerOpts...)
+	quote, err := a.RUsecase.GetOptimalQuote(ctx, *tokenIn, tokenOutDenom, req.SwapMethod(), routerOpts...)
```

Comment:
Instead of passing in the swap method as a parameter, would it be possible to create a wrapper around `GetOptimalQuote` such as `GetOptimalQuoteInGivenOut` that would call `GetOptimalQuote` under the hood and reformat the result into the final type?

The reason I'm suggesting this is that passing in an argument to control the execution flow is a code smell that adds complexity.

By having a separate function, we would be able to functionally decouple the "in given out" implementation while also avoiding any changes to the core `GetSplitQuote`

Reply:
Thanks for the feedback Roman, I have implemented [GetOptimalQuoteInGivenOut](https://github.com/osmosis-labs/sqs/pull/397/files#diff-9d1423fc5abbd62d4ff5f8151c5277b0dbc5af89a2e603f2b0c211835f690a11R216)!  :rocket: